backend/server/game_api.js

```diff
@@ -0,0 +1,637 @@
+                return res.status(400).json({ msg: 'User is not a player in the game' });
+            }
+
+            if (start_row > gameDetails.dimension || start_col > gameDetails.dimension || end_col > gameDetails.dimension || end_row > gameDetails.dimension) {
+                return res.status(400).json({ msg: 'Ship is not inside board borders' });
+            };
+
+            if (start_row < 1 || start_col < 1 || end_col < 1 || end_row < 1) {
+                return res.status(400).json({ msg: 'Ship is not inside board borders' });
+            };
+
+            const canPlace = await checkShips(req, gameDetails.dimension, ship_size);
+            if (!canPlace) {
+                return res.status(400).json({ msg: 'There are no more ships of this size to place' });
+            }
+
+
+            const ships = await pool.query('select * from ships where game_id = $1 and user_id = $2', [req.params.game_id, req.user.id]);
+            if (ships.rows.length === 0) {
+                await placeShip(req.params.game_id, req.user.id, ship_size, start_row, start_col, end_row, end_col, res);
+            } else {
+                const shipPlacementResult = ships.rows.map(ship => checkShipPlacement(start_row, end_row, start_col, end_col, ship)).reduce((accumulator, currentValue) => accumulator + currentValue, 0);
+                if (shipPlacementResult > 0) {
+                    return res.status(400).json({ msg: 'Ship cannot be next to another ship' });
+                } else {
+                    await placeShip(req.params.game_id, req.user.id, ship_size, start_row, start_col, end_row, end_col, res);
+                }
+            }
+        } else {
+            return res.status(400).json({ msg: 'Game is not in correct state or user not correct user' });
+        }
+
+    } catch (e) {
+        res.status(500).json({ msg: 'Server error' });
+    }
+
+});
+
+
+
+// Unplace a ship
+gameRouter.delete('/games/:game_id/place', async (req, res) => {
+    const { ship_size, start_row, start_col, end_row, end_col } = req.body;
+
+    try {
+        const game = await pool.query('select * from games where id = $1', [req.params.game_id]);
+        const gameDetails = game.rows[0];
+
+        if (gameDetails.state === 'accepted' || (gameDetails.user1 === req.user.id && gameDetails.state === 'user2_ready') || (gameDetails.user2 === req.user.id && gameDetails.state === 'user1_ready')) {
+            const userShips = await pool.query('select * from ships where game_id = $1 and user_id = $2', [req.params.game_id, req.user.id]);
+            const check = userShips.rows.map(dbShip => {
+                if (dbShip.size === ship_size && dbShip.start_row === start_row && dbShip.start_col === start_col && dbShip.end_row === end_row && dbShip.end_col === end_col) {
+                    return true;
+                }
+            });
+            if (check.includes(true)) {
```

Comment:
Instead of `map` and `includes` you can use `some` which checks if there is at least one element in the list that passes some test. 
See the explanation of this function here:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/some